airbyte-integrations/connectors/source-amazon-ads/source_amazon_ads/spec.yaml

```diff
@@ -98,6 +98,18 @@ connectionSpecification:
       type: array
       uniqueItems: true
       order: 9
+    report_record_types:
+      title: Report Record Types
+      description:
+        Optional configuration which accepts an array of string of record types.
+        Leave blank for default behaviour to pull all report types.
+        Use this config option only if you want to pull specific report type(s).
+        See <a href="https://advertising.amazon.com/API/docs/en-us/reporting/v2/report-types">docs</a>
+        for more details
+      order: 10
+      type: array
+      items:
+        type: string
```

Comment:
Make it `enum` just like `state_filter`. Then, it will be easier for the users to pick predefined values from the list and prevent them from entering invalid ones.

**NB!** There are `asins_keywords` and `asins_targets` for Sponsored Products. Make sure that they both work. Add test cases for `asins`, `asins_keywords`, and `asins_targets`.

Reply:
since all report types are not supported by each category of reports, making them enum doesn't make sense.

Also, this would be an optional config option, just like `profiles`

Reply:
Now, any value can be entered, even nonexistent in any report type. Enum would limit possible values to those that exist at least in one report type. Anyway, your code now ignores invalid values so that nothing would change on the code side.